tests: Add some tests for Yelp

pwithnall opened the pull request:
These cover basic Yelp functionality, plus one of our key additions (the
gnome-shell search provider).

Signed-off-by: Philip Withnall <withnall@endlessm.com>

https://phabricator.endlessm.com/T28834

pwithnall commented:
Example of a passing test run here: https://openqa.endlessm.com/tests/57022

wjt commented on needles/yelp_search_provider_results.json:
```diff
@@ -0,0 +1,15 @@
+  ],
+  "area": [
+    {
+      "": "Whole screen, minus shell bottom bar (changes between users; clock changes)",
```
The screenshot to go with this needle does not include a Yelp search result.

I don't think we should be matching the whole screen – I think we should be matching just the search result.

wjt replied:
"Search Google for ____" is not the Yelp provider.

pwithnall replied:
The perils of coming back to OpenQA after context-switching. I added that screenshot as a placeholder and never got back to it. (I know “Search Google for x” is not a Yelp search result.) Ooops :(

pwithnall commented:
Fixed the incorrect search provider results needle and added a second commit to improve the desktop matching (unrelated to Yelp but generally useful): https://openqa.endlessm.com/tests/57190#step/yelp/1

wjt commented on tests/yelp.pm:
```diff
@@ -0,0 +1,40 @@
+
+    # Search for a help topic, and results should be shown by a search provider.
+    type_very_safely('add user');
+    assert_screen('yelp_search_provider_results', 10);
```
Might be nice to match one of the results, click on it, and check it works, but that could be something for the future.
